src/code/z_skin_matrix.c

```diff
@@ -1,42 +1,692 @@
+    m1[16 + 1] = temp & 0xFFFF;
+
+    temp = src->xz * 0x10000;
+    m1[2] = (temp >> 0x10);
+    m1[16 + 2] = temp & 0xFFFF;
+
+    temp = src->xw * 0x10000;
+    m1[3] = (temp >> 0x10);
+    m1[16 + 3] = temp & 0xFFFF;
+
+    temp = src->yx * 0x10000;
+    m1[4] = (temp >> 0x10);
+    m1[16 + 4] = temp & 0xFFFF;
+
+    temp = src->yy * 0x10000;
+    m1[5] = (temp >> 0x10);
+    m1[16 + 5] = temp & 0xFFFF;
+
+    temp = src->yz * 0x10000;
+    m1[6] = (temp >> 0x10);
+    m1[16 + 6] = temp & 0xFFFF;
+
+    temp = src->yw * 0x10000;
+    m1[7] = (temp >> 0x10);
+    m1[16 + 7] = temp & 0xFFFF;
+
+    temp = src->zx * 0x10000;
+    m1[8] = (temp >> 0x10);
+    m1[16 + 8] = temp & 0xFFFF;
+
+    temp = src->zy * 0x10000;
+    m1[9] = (temp >> 0x10);
+    m2[9] = temp & 0xFFFF;
+
+    temp = src->zz * 0x10000;
+    m1[10] = (temp >> 0x10);
+    m2[10] = temp & 0xFFFF;
+
+    temp = src->zw * 0x10000;
+    m1[11] = (temp >> 0x10);
+    m2[11] = temp & 0xFFFF;
+
+    temp = src->wx * 0x10000;
+    m1[12] = (temp >> 0x10);
+    m2[12] = temp & 0xFFFF;
+
+    temp = src->wy * 0x10000;
+    m1[13] = (temp >> 0x10);
+    m2[13] = temp & 0xFFFF;
+
+    temp = src->wz * 0x10000;
+    m1[14] = (temp >> 0x10);
+    m2[14] = temp & 0xFFFF;
+
+    temp = src->ww * 0x10000;
+    m1[15] = (temp >> 0x10);
+    m2[15] = temp & 0xFFFF;
+}
+
+Mtx* func_800A7E70(GraphicsContext* gfxCtx, MtxF* src) {
+    Mtx* displayList = Graph_Alloc(gfxCtx, 0x40);
+
+    if (displayList == 0) {
+        osSyncPrintf("Skin_Matrix_to_Mtx_new() 確保失敗:NULLを返して終了\n", displayList);
+        return 0;
+    }
+    SkinMatrix_MtxFToMtx(src, displayList);
+    return displayList;
+}
+
+// Rotation matrix rotates by angle a around the vector with components (x,y,z)
+/*
+My impression is that this is functionally correct but I cannot be 100% sure of
+equivalence because diff output is not very close to matching.
+*/
+#ifdef NON_EQUIVALENT
+void func_800A7EC0(MtxF* mf, s16 a, f32 x, f32 y, f32 z) {
+    f32 sin;
+    f32 cos;
+    f32 ct;
+    f32 xx;
+    f32 yy;
+    f32 zz;
+    f32 xy;
+    f32 xz;
+    f32 yz;
+    f32 sx;
+    f32 sy;
+    f32 sz;
+
+    sin = Math_Sins(a);
+    cos = Math_Coss(a);
+
+    ct = 1.0f - cos;
+    sx = sin * x;
+    sy = sin * y;
+    sz = sin * z;
+
+    xx = x * x;
+    xy = x * y;
+    xz = x * z;
+    yy = y * y;
+    yz = y * z;
+    zz = z * z;
+
+    mf->xx = (f32)ct * xx + cos;
+    mf->xy = (f32)((ct * xy) + sz);
+    mf->xz = (f32)((ct * xz) - sy);
+    mf->xw = 0.0f;
+
+    mf->yx = (f32)((ct * (xy)) - sz);
+    mf->yy = (f32)ct * yy + cos;
+    mf->yz = (f32)((ct * (yz)) + sx);
+    mf->yw = 0.0f;
+
+    mf->zx = (f32)((ct * (xz)) + sy);
+    mf->zy = (f32)((ct * (yz)) - sx);
+    mf->zz = (f32)ct * zz + cos;
+    ;
+    mf->zw = 0.0f;
+
+    mf->wx = 0.0f;
+    mf->wy = 0.0f;
+    mf->wz = 0.0f;
+    mf->ww = 1.0f;
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A7EC0.s")
+#endif
+
+// unknown
```

Comment:
we'll assume that anything not documented is not understood yet, can take this out imo